dockers/docker-sonic-telemetry/certificate_rollover_checker

```diff
@@ -0,0 +1,136 @@
+    telemetry_table_keys = telemetry_table.getKeys()
+    if "certs" in telemetry_table_keys:
+        certs_info = dict(telemetry_table.get("certs")[1])
+        if "server_crt_acms" in certs_info and "server_key_acms" in certs_info:
+            certificate_path = certs_info["server_crt_acms"]
+            private_key_path = certs_info["server_key_acms"]
+            syslog.syslog(syslog.LOG_INFO, "Path of certificate file is '{}'".format(certificate_path))
+            syslog.syslog(syslog.LOG_INFO, "Path of key file is '{}'".format(private_key_path))
+        else:
+            syslog.syslog(syslog.LOG_ERR,
+                          "Failed to retrieve the path of certificate and key file from 'TELEMETRY' table!")
+            sys.exit(2)
+    else:
+        syslog.syslog(syslog.LOG_ERR,
+                      "Failed to retrieve the certificate information from 'TELEMETRY' table!")
+        sys.exit(3)
+
+    while True:
+        if not os.path.exists(certificate_path) or not os.path.exists(private_key_path):
+            syslog.syslog(syslog.LOG_ERR,
+                          "Certificate or key file did not exist on device and sleep '{}' seconds to check again ...".format(CERTIFICATE_CHECKING_INTERVAL_SECS))
+            time.sleep(CERTIFICATE_CHECKING_INTERVAL_SECS)
+        else:
+            break
+
+    certificate_last_mod_time = get_file_last_mod_time(certificate_path)
+    private_key_last_mod_time = get_file_last_mod_time(private_key_path)
+
+    while True:
+        certificate_mod_time = get_file_last_mod_time(certificate_path)
+        private_key_mod_time = get_file_last_mod_time(private_key_path)
+        syslog.syslog(syslog.LOG_INFO,
+                      "Last modification time of certificate file is: '{}'".format(time.ctime(certificate_last_mod_time)))
+        syslog.syslog(syslog.LOG_INFO,
+                      "Last modification time of key file is: '{}'".format(time.ctime(private_key_last_mod_time)))
+
+        if (certificate_mod_time > certificate_last_mod_time
+           or private_key_mod_time > private_key_last_mod_time):
+            syslog.syslog(syslog.LOG_INFO,
+                          "Last modification time of certificate file is changed to '{}': ".format(time.ctime(certificate_mod_time)))
+            syslog.syslog(syslog.LOG_INFO,
+                          "Last modification time of key file is changed to '{}': ".format(time.ctime(private_key_mod_time)))
+            syslog.syslog(syslog.LOG_INFO,
+                          "Secrets were rolled over and restarting streaming telemetry service ...")
+            restart_streaming_telemetry()
+
+        # Wait for specified seconds and then do the next round checking
+        syslog.syslog(syslog.LOG_INFO,
+                      "Sleeping '{}' seconds before doing the next round rollover checking ...".format(CERTIFICATE_CHECKING_INTERVAL_SECS))
+        time.sleep(CERTIFICATE_CHECKING_INTERVAL_SECS)
+
+
+def main():
+    certificate_rollover_check()
```

Comment:
>certificate_rollover_check [](http://example.com/codeflow?start=4&length=26)

Could you try inotify ([https://www.linuxjournal.com/content/linux-filesystem-events-inotify](https://www.linuxjournal.com/content/linux-filesystem-events-inotify) ) instead of reinvent the wheel?

Reply:
Updated.